x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/physical/local/ReplaceSourceAttributes.java

```diff
@@ -42,9 +42,26 @@ protected PhysicalPlan rule(EsSourceExec plan) {
             if (tsid == null || timestamp == null) {
                 throw new IllegalStateException("_tsid or @timestamp are missing from the time-series source");
             }
-            return new EsQueryExec(plan.source(), plan.index(), plan.indexMode(), List.of(docId, tsid, timestamp), plan.query());
+            MetadataAttribute scoring = getScoringAttribute(plan);
+            List<Attribute> attributes = scoring != null ? List.of(docId, tsid, timestamp, scoring) : List.of(docId, tsid, timestamp);
+            return new EsQueryExec(plan.source(), plan.index(), plan.indexMode(), attributes, plan.query());
         } else {
-            return new EsQueryExec(plan.source(), plan.index(), plan.indexMode(), List.of(docId), plan.query());
+            MetadataAttribute scoring = getScoringAttribute(plan);
```

Comment:
I think we can get the `scoring` attribute out of the if/else branching and also the return (we can init a list outside the if/else, add whatever attribute we need to into the if/else, return outside), to avoid duplication and make the code more legible.